File: src/tests/fidl/source_compatibility/bits-member-add/dart/step_00_before.dart
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// TODO(https://fxbug.dev/84961): Fix null safety and remove this language version.
// @dart=2.9

import 'package:fidl_fidl_test_bitsmemberadd/fidl_async.dart' as fidllib;

// [START contents]
void useBits(fidllib.Flags bits) {
  if ((bits & fidllib.Flags.optionA).$value != 0) {
    print('option A is set');
  }
  if ((bits & fidllib.Flags.optionB).$value != 0) {
    print('option B is set');
  }
  if (bits.hasUnknownBits()) {
    print('unknown options: ${bits.getUnknownBits()}');
  }
}
// [END contents]

File: src/tests/fidl/source_compatibility/bits-member-add/rust/step_00_before.rs
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#![allow(dead_code)]

use fidl_fidl_test_bitsmemberadd as fidl_lib;

// [START contents]
fn use_bits(bits: &fidl_lib::Flags) {
    if bits.contains(fidl_lib::Flags::OPTION_A) {
        println!("option A is set");
    }
    if bits.contains(fidl_lib::Flags::OPTION_B) {
        println!("option B is set");
    }
    if bits.has_unknown_bits() {
        println!("unknown options: {:x}", bits.get_unknown_bits());
    }
}
// [END contents]

fn main() {}
